src/Utilities/TagsModal.test.js

```diff
@@ -143,6 +143,12 @@ describe('TagsModal', () => {
   });
 
   describe('API', () => {
+    const mockSetState = jest.fn();
+
+    jest.mock('react', () => ({
+      useState: (initial) => [initial, mockSetState],
+    }));
```

Comment:
I am not sure it's a good idea to mock react components/functions/exports anyhow. This way, you are setting `react` to return only useState with this custom fixture, whereas other methods like useEffect, useMemo, whatever `react` contains is now set to zero. This can be a reason why you are unable to write more advanced tests.

Reply:
I'm not sure I understand anything you are saying here. :)